### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by Furgas**

### CODE REVIEW

File: Dreamscapes/Ldap/Core/Ldap.php [hunk 1]
```diff
@@ -714,7 +719,7 @@ public function startTls()
      */
     public function unbind()
     {
-        ldap_unbind($this->resource);
+        @ldap_unbind($this->resource);
```
Comment (Alaneor) [inline Dreamscapes/Ldap/Core/Ldap.php]: I am hesitant to supress PHP errors/warnings in situations where I do not call `$this->verifyOperation()` to check that the operation was successful. Are you seeing actual warnings/errors in your console while calling `Ldap::unbind()`?

I propose that either we remove the silencer or we add a call to `$this->verifyOperation()` below. Perhaps the second option would be best...

File: Dreamscapes/Ldap/Core/Ldap.php [hunk 2]
```diff
@@ -752,11 +757,10 @@ public function __call($method, $args)
         return call_user_func_array([$this, 'ldapSearch'], $args);
     }
 
-
     /**
      * Check the ldap status code and throw exception on error
      *
-     * @return void
```
Comment (Alaneor) [inline Dreamscapes/Ldap/Core/Ldap.php]: But it does not return anything... How do document a function that does not return anything? Also, for consistency and for being explicit, I prefer to include a `@return` docblock for all methods.

File: Dreamscapes/Ldap/Core/Ldap.php [hunk 3]
```diff
@@ -301,10 +301,12 @@ public function add($dn, array $entry)
...
     }
 
-    /** Alias of self::unbind() */
+    /**
+     * Alias of self::unbind()
+     */
     public function close()
     {
-        return $this->unbind();
+        $this->unbind();
```
Comment (Alaneor) [inline Dreamscapes/Ldap/Core/Ldap.php]: The `Ldap::close()` method is an alias for `Ldap::unbind()`. The implementation is done in the `unbind()` method. If I for whatever reason decide to change the return value of this method, I want the alias to reflect that change. `Ldap::unbind()` currently returns `void`, so this method returns also `void`.

Please revert this change.

File: Dreamscapes/Ldap/Core/Ldap.php [hunk 4]
```diff
@@ -752,11 +757,10 @@ public function __call($method, $args)
         return call_user_func_array([$this, 'ldapSearch'], $args);
     }
 
-
```
Comment (Alaneor) [inline Dreamscapes/Ldap/Core/Ldap.php]: I always separate methods' visibility change with two empty lines. E.g. first there are public methods, <2 lines>, protected methods, <2 lines>, private methods. It helps me visually distinguish these groups.

Please revert this change.

**Comment by Alaneor** [comment]
Hi, thanks for the PR! I added some inline comments to your proposed changes. Once we work those out, I will be happy to merge.

File: Dreamscapes/Ldap/Core/Ldap.php (hunk 1 above)
Comment (Furgas) [inline Dreamscapes/Ldap/Core/Ldap.php]: I think, you can't do anything with LDAP resource whether unbind failed or not. So I personally just ignore anything coming from this method. The only reason I could think of when it would be useful is to catch some logic errors when trying to unbind previously unbinded link. Throwing exception would be too much IMHO, so I would revert the change or leave it - tell me what's yout decision.

File: Dreamscapes/Ldap/Core/Ldap.php (hunk 2 above)
Comment (Furgas) [inline Dreamscapes/Ldap/Core/Ldap.php]: OK. As it's a subjective preference I should honor your decision as main author:) Commit on the way.

File: Dreamscapes/Ldap/Core/Ldap.php (hunk 1 above)
Comment (Alaneor) [inline Dreamscapes/Ldap/Core/Ldap.php]: Okay, that sounds logical... Calling `ldap_unbind()` on unbound resource seems to only cause a warning, so let's keep it as is - with silencer and without throwing anything.

File: Dreamscapes/Ldap/Core/Ldap.php (hunk 3 above)
Comment (Furgas) [inline Dreamscapes/Ldap/Core/Ldap.php]: Personally I would then update `close()` method accordingly. You will still need to do that if you decide to add some arguments to the method. Please confirm.

Comment (Alaneor) [inline Dreamscapes/Ldap/Core/Ldap.php]: That is true. The most likely update I can think of would be returning something to indicate the success/failure of the unbinding, and for that the `return` can take care of it for me. I insist.:) Thank you!

**Comment by Furgas** [comment]
Wait a moment, I'll try to clean commits.

File: Dreamscapes/Ldap/Core/Ldap.php (hunk 3 above)
Comment (Furgas) [inline Dreamscapes/Ldap/Core/Ldap.php]: OK.

File: Dreamscapes/Ldap/Core/Ldap.php (hunk 4 above)
Comment (Furgas) [inline Dreamscapes/Ldap/Core/Ldap.php]: OK.

**Comment by Furgas** [comment]
Cleaned up.

**Comment by Alaneor** [comment]
Excellent, thanks!

This is now merged in develop branch. I will likely release this as 3.1.0 sometime during the evening.

**Comment by Furgas** [comment]
This was just preliminary work, more like patch (3.0.1). My bigger goal is to add better support for paged searching (backward compatible). I would wait for that with 3.1.0 release. Of course, if you choose to merge it.

**Comment by Alaneor** [comment]
Okay, sounds good to me.

What do you have in mind in terms of paging support? My initial goal for this library was only to provide an object model around PHP's ldap functions and I deliberately avoided adding extra functionality on top of it. The intention was to make interacting with ldap testable (by mocking the underlying ldap interface).

I am just pointing this out in case you plan on adding extra features that have no direct connection to what PHP provides - that would be out of scope for this library.

[I did this once](https://github.com/Alaneor/Ldap) (adding "custom" features to a library like this), but realised that it killed the "testability" part. The originally original motivation to write this was to be able to write tests for [AD-X](https://github.com/Alaneor/AD-X), which is a feature-rich Ldap library, but I no longer have ambitions to properly implement this into it and continue with writing the tests.

**Comment by Furgas** [comment]
I will try to follow the scope you mention. I need it, so I will do it nevertheless. Of course it's totally up to you if you decide to merge it.
